### ISSUE TITLE
UTC TIME values in CSV outputs step 3 missing 'Z'

### ISSUE THREAD

**Issue by lbesnard** [aodn/content#297]
It seems like a recurrent question we receive at info@emii.org.au (cf @kereid ) asking if the time values are UTC. And indeed, the csv output is not self explanatory as the time values are always written without the 'Z' indicating the time is UTC such as "2017-10-02T07:30:00".

It would be good to investigate the way to add this information in the CSV output, probably via geoserver.

@jonescc would you have an idea regarding this issue ?

**Comment by pblain** [comment]
Merged.

**Comment by mhidas** [comment]
For consistency, should we be applying the same date/time output format in the standard CSV output type, not just the one with our custom header?

And just to clarify, is this new formatting adding the timezone information contained in the WFS values returned (which will be 'Z'/UTC most of the time, but not guaranteed), or is it simply slapping a 'Z' on the end of whatever the timestamp is? (question for @pmbohm I guess)

We did develop the habit of always specifying UTC as the output timezone in all _map and _data views, which means they are returned by the database as type `timestamp without time zone`. However I've realised I forgot to do this in some recently created views, so there may be some `timestamp with time zone` cases as well.

**Comment by pmbohm** [comment]
@mhidas Good questions
Regarding CSV output, apparently for our servers we always will use CSV with headers. 

Regarding datasets with or without timezone; Can you test this? 
The assumption was made that all dates are UTC.

The code may or not do the right thing as it deals with the results as proper dates. Below is the parsing function used, but how it will handles the passed in `timestamp with time zone` date object (as interpreted by Geoserver from the DB) is not known by me.

```
    /**
     * Constructs an instance from an Object that represents a datetime,
     * forcing the time zone to that specified.
     * <p>
     * If the object implies a chronology (such as GregorianCalendar does),
     * then that chronology will be used, but with the time zone adjusted.
     * Otherwise, ISO is used in the specified time zone.
     * If the specified time zone is null, the default zone is used.
     * Thus if a GregorianCalendar is passed in, the chronology used will
     * be GJ, but if a Date is passed in the chronology will be ISO.
     * <p>
     * The recognised object types are defined in
     * {@link org.joda.time.convert.ConverterManager ConverterManager} and
     * include ReadableInstant, String, Calendar and Date.
     * The String formats are described by {@link ISODateTimeFormat#dateTimeParser()}.
     *
     * @param instant  the datetime object, null means now
     * @param zone  the time zone, null means default time zone
     * @throws IllegalArgumentException if the instant is invalid
     */
    public DateTime(Object instant, DateTimeZone zone) {
        super(instant, zone);
    }
```

**Comment by mhidas** [comment]
As far as I can tell, the changes introduced in https://github.com/aodn/geoserver-build/pull/261 are in prod now. I did a quick comparison between timestamps in file attribute, in the db views and in the CSV (csv-with-metadata-header format, downloaded from geoserver-123 via Layer Preview), and they are all the same!

What I couldn't test, however, is whether this is dependent on the default timezone set in the db server. I think back when we first started creating these layers the db server was using local time, and columns left as `timestamp with time zone` in the views would then end up getting to geoserver in local time (and then labelled as UTC). I think this is why we adopted the practice of always explicitly converting timestams to UTC in the views.
